src/components/VariableCreateModal.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+  <p-modal v-model:showModal="internalValue" :title="localization.info.newVariable">
+    <p-form @submit="submit">
+      <p-content>
+        <p-label :label="localization.info.name" :state="nameState" :message="nameErrorMessage">
+          <p-text-input v-model="name" :state="nameState" />
+        </p-label>
+
+        <p-label :label="localization.info.value" :state="valueState" :message="valueErrorMessage">
+          <p-text-input v-model="value" :state="valueState" />
+        </p-label>
+
+        <p-label :label="localization.info.tags">
+          <p-tag-input v-model="tags" />
+        </p-label>
+      </p-content>
+    </p-form>
+
+    <template #actions>
+      <p-button :loading="pending" @click="submit">
+        Create
+      </p-button>
+    </template>
+    <template #cancel>
+      <p-button inset @click="internalValue = false">
+        Close
+      </p-button>
+    </template>
+  </p-modal>
+</template>
+
+<script lang="ts" setup>
+  import { showToast } from '@prefecthq/prefect-design'
+  import { useValidation, useValidationObserver, ValidationRule } from '@prefecthq/vue-compositions'
+  import { isNull } from 'lodash'
+  import { computed, ref } from 'vue'
+  import { useWorkspaceApi } from '@/compositions'
+  import { localization } from '@/localization'
+  import { VariableCreate } from '@/models'
+  import { isRequired, isString, isValidIf } from '@/utilities'
+
+  const props = defineProps<{
+    showModal: boolean,
+  }>()
+
+  const emits = defineEmits<{
+    (event: 'update:showModal', value: boolean): void,
+  }>()
+
+  const internalValue = computed({
+    get() {
+      return props.showModal
+    },
+    set(value: boolean): void {
+      emits('update:showModal', value)
+    },
+  })
+
+  const api = useWorkspaceApi()
+
+  const isUnique = isValidIf(async (value) => {
+    if (isNull(value) || !isString(value)) {
+      return false
+    }
+    const variable = await api.variables.getVariableByName(value)
```

Comment:
Since validation is triggered every time the value changes you might want to debounce/cancel this. 

There's an example in automations where it denounced depending on the source. See the isValidTemplate method. 

https://github.com/PrefectHQ/nebula-ui/blob/main/src/components/AutomationActionSendNotification.vue

Reply:
Great minds - I pushed a version of that after I caught that on self-review lol - let me double check it matches roughly what you've already done

Reply:
Ah ok I don't think just a debounced ref makes sense, I'll write one that uses the validation factory methods

Reply:
A denounced ref is a good idea. It would be simpler than what I have. But I think that would have the possibility of passing validation before the denounces ref gets updated. 

But you could validate the denounced ref during normal validation and manually validate the actual ref during submission. Not sure if that would work correctly. 

The method i wrote debounces validation unless an observer is doing the validation then it fires right away.